The first change is needed to print on HP DJ690C.
The second reduces output size a little.

--- espgs-7.05.6/src/gdevcd8.c.orig	2002-07-30 20:53:21.000000000 +0200
+++ espgs-7.05.6/src/gdevcd8.c	2003-07-02 22:06:12.000000000 +0200
@@ -1847,7 +1847,7 @@
 	}
 	/* Skip blank lines if any */
 	if (num_blank_lines > 0) {
-	    fprintf(prn_stream, "\033*b%dY", num_blank_lines / (cdj850->yscal + 1));
+	    fprintf(prn_stream, "\033*b%dy", num_blank_lines / (cdj850->yscal + 1));
 	    memset(data_ptrs->plane_data[0][0], 0,
 		   (misc_vars->plane_size * 2 * misc_vars->num_comps));
 	    memset(data_ptrs->plane_data_c[0][0], 0,
@@ -1883,9 +1883,11 @@
     int out_count = gdev_pcl_mode9compress(plane_size, curr, prev, out_data);
 
     /* and output the data */
-	fprintf(prn_stream, "%d%c", out_count, plane_code);
     if (out_count > 0) {
+	fprintf(prn_stream, "%d%c", out_count, plane_code);
 	fwrite(out_data, sizeof(byte), out_count, prn_stream);
+   } else {
+      putc(plane_code, prn_stream);
     }
 }
 
